### ISSUE TITLE
ZeroDivisionError happened when nuc_data_make

### ISSUE THREAD

**Issue by zxkjack123** [pyne/pyne#983]
Python version: 2.7
Error information:
```
Traceback (most recent call last):
  File "./scripts/nuc_data_make", line 6, in <module>
    main()
  File "/home/zxk/.local/lib/python2.7/site-packages/pyne/dbgen/nuc_data_make.py", line 157, in main
    make_map[mo](args)
  File "/home/zxk/.local/lib/python2.7/site-packages/pyne/dbgen/decay.py", line 472, in make_decay
    make_decay_half_life_table(nuc_data, build_dir)
  File "/home/zxk/.local/lib/python2.7/site-packages/pyne/dbgen/decay.py", line 380, in make_decay_half_life_table
    level_list = parse_level_data(build_dir)
  File "/home/zxk/.local/lib/python2.7/site-packages/pyne/dbgen/decay.py", line 252, in parse_level_data
    level_list = ensdf.levels(f, level_list)
  File "/home/zxk/.local/lib/python2.7/site-packages/pyne/ensdf.py", line 1008, in levels
    _parse_level_record(level_l)
  File "/home/zxk/.local/lib/python2.7/site-packages/pyne/ensdf.py", line 226, in _parse_level_record
    tfinal, tfinalerr = _to_time(l_rec.group(5), l_rec.group(6))
  File "/home/zxk/.local/lib/python2.7/site-packages/pyne/ensdf.py", line 137, in _to_time
    tfinal, tfinalerr = _halflife_to_seconds(value, err, t_unit)
  File "/home/zxk/.local/lib/python2.7/site-packages/pyne/ensdf.py", line 118, in _halflife_to_seconds
    sec_err = (scale / (value - minus) - sec_time,
ZeroDivisionError: float division by zero
```

**Comment by zxkjack123** [comment]
I find that this happens when at this function:`_halflife_to_seconds(value, err, units)` of `pyne/ensdf.py`.
While the parameters are:
`_halflife_to_secondes(0.15, 0.15, MeV)`.

Here, `value` equals to `err`, causes the `(value - minus)` equals to 0, thus ZeroDivisonError happens.

Suggestion: check the divisor before the division.

**Comment by kkiesling** [comment]
I am receiving the same error on a clean docker build (ubuntu 16.04) when `scripts/nuc_data_make` is run. My suspicion is this error was introduced in PR #961 (though wouldn't it have shown up in other peoples' builds?). It doesn't look like it would have been caught by CI since it comes at the nuc_data_make step.

@zxkjack123 I was able to get around this error by using PyNE version 0.5.10.

**Comment by kkiesling** [comment]
Pinging @pkienzle on this issue, since I believe this error was introduced in PR #961. It would not have been caught in CI though since CI does not run ./scripts/nuc_data_make.

**Comment by scopatz** [comment]
Hi All, I am a little confused.  Is this an issue on develop or not? Thanks!

**Comment by kkiesling** [comment]
Yes, it is on the develop branch. The error appears though when running nuc_data_make.

**Comment by scopatz** [comment]
Seems like this should be a simple fix.  Are either of you willing to put in a PR? I'd really appreciate it!

**Comment by kkiesling** [comment]
I don't think it actually is that easy of a fix. There is the simple "check if it is zero and make it something else" to stop creating an error, but we (@gonuke and others) just tried to figure out what that value should be and are unsure. I think it would be appropriate for @pkienzle to provide some guidance on this as the physics and understanding of the data behind it seem to be a bit more in depth.

**Comment by pkienzle** [comment]
Yes, the interpretation is challenging.  Since energy cannot be negative, then 0.15 MeV +/- 0.15 MeV clearly cannot represent a gaussian uncertainty.  Furthermore, the 1/x transformation drives the half-life to infinity as energy goes to zero.

Doing a bit of simulation (sampling from a gaussian of the given width in energy truncated below zero and histogramming the converted result), the current scheme works well for uncertainty small relative to energy.  Large uncertainties sample disproportionately from higher energies, with the histogram of log(time) peaked at the time corresponding to the uncertainty.  For example, the peak for ```0.15 +/- 15 MeV``` is sitting at ```3E-23 s``` corresponding to ```15 MeV``` rather than ```3E-21 s``` corresponding to ```0.15 MeV```.  The histogram drops rapidly below ```3E-23 s```.

Assuming that the order of magnitude of the measurement is correct, I suggest keeping the current lower bound as ```converted(value+plus)```, but setting the upper bound as ```converted(max(value/10, value-minus))```.  That is, change line 118 in ensdf.py as follows:
```
        sec_err = (scale / max(0.1*value, value - minus) - sec_time,
                   sec_time - scale / (value + plus))
```
This captures the majority of the density of the truncated distribution while respecting the nominal value and giving a generous upper bound.  The same holds even when plus/minus correspond to 3-sigma rather than 1-sigma intervals.

A test case for 0.15 +/- 0.15 MeV should be added to tests/test_ensdf.py, giving:

    3.0415850554917978e-21, (2.7374265499426181e-20, 1.5207925277458989e-21)

Sorry, can't test it myself since I haven't figured out how to build on mac and I have no linux box handy.

**Comment by scopatz** [comment]
I am able to reproduce this.  It is quite the problem now because the conda packages are broken (due to MOAB) and we can't regenerate them due in part to this issue.

**Comment by scopatz** [comment]
@pkienzle - your fix seems to work! Thanks!

**Comment by scopatz** [comment]
Fix is available in #1005
